pkg/awsutils/awsutils.go

```diff
@@ -602,90 +608,96 @@ func (cache *EC2InstanceMetadataCache) getENIMetadata(eniMAC string) (ENIMetadat
 		deviceNum = 0
 	}
 
-	log.Debugf("Found ENI: %s, MAC %s, device %d", eniID, eniMAC, deviceNum)
-
-	// Get IPv4 and IPv6 addresses assigned to interface
-	cidr, err := cache.imds.GetSubnetIPv4CIDRBlock(ctx, eniMAC)
-	if err != nil {
-		awsAPIErrInc("GetSubnetIPv4CIDRBlock", err)
-		return ENIMetadata{}, err
-	}
-
-	imdsIPv4s, err := cache.imds.GetLocalIPv4s(ctx, eniMAC)
-	if err != nil {
-		awsAPIErrInc("GetLocalIPv4s", err)
-		return ENIMetadata{}, err
-	}
-
-	ec2ip4s := make([]*ec2.NetworkInterfacePrivateIpAddress, len(imdsIPv4s))
-	for i, ip4 := range imdsIPv4s {
-		ec2ip4s[i] = &ec2.NetworkInterfacePrivateIpAddress{
-			Primary:          aws.Bool(i == 0),
-			PrivateIpAddress: aws.String(ip4.String()),
-		}
-	}
+	log.Debugf("Found ENI: %s, MAC %s, device %d, network card %d", eniID, eniMAC, deviceNum, networkCard)
 
+	var subnetV4Cidr string
+	var ec2ip4s []*ec2.NetworkInterfacePrivateIpAddress
 	var ec2ip6s []*ec2.NetworkInterfaceIpv6Address
 	var subnetV6Cidr string
-	if cache.v6Enabled {
-		// For IPv6 ENIs, do not error on missing IPv6 information
-		v6cidr, err := cache.imds.GetSubnetIPv6CIDRBlocks(ctx, eniMAC)
-		if err != nil {
-			awsAPIErrInc("GetSubnetIPv6CIDRBlocks", err)
-		} else {
-			subnetV6Cidr = v6cidr.String()
-		}
+	var ec2ipv4Prefixes []*ec2.Ipv4PrefixSpecification
+	var ec2ipv6Prefixes []*ec2.Ipv6PrefixSpecification
 
-		imdsIPv6s, err := cache.imds.GetIPv6s(ctx, eniMAC)
+	if networkCard == 0 {
+		// Get IPv4 and IPv6 addresses assigned to interface
```

Comment:
Need to remove IPv6 from the comment

Reply:
@jayanthvn we are fetching both IPv4 and IPv6 addresses for ENI's on network card 0. Comment is same as earlier to this change. Github side by side comparison is a bit messed for this PR